src/test/java/uk/gov/pay/api/it/validation/PaymentResourceSearchValidationITest.java

```diff
@@ -166,6 +172,31 @@ public void searchPayments_errorWhenFromAndToDatesAreNotInZoneDateTimeFormat() t
                 .assertThat("$.code", is("P0401"))
                 .assertThat("$.description", is("Invalid parameters: from_date, to_date. See Public API documentation for the correct data formats"));
     }
+    
+    @Test
+    public void searchPayments_errorWhenCardDigitsAreInvalid() throws Exception {
+        InputStream body = searchPayments(API_KEY,
+                ImmutableMap.<String, String>builder()
+                        .put("reference", VALID_REFERENCE)
+                        .put("email", VALID_EMAIL)
+                        .put("state", VALID_STATE)
+                        .put("first_digits_card_number", "4242")
+                        .put("last_digits_card_number", "423")
```

Comment:
Maybe have one of them be the wrong length and the other one be the right length but with non-digits in it to cover a broader swathe of the validation

Reply:
👍